### DISCUSSION TITLE
To remove Counters from the main FleekERC721 contract

### DISCUSSION THREAD

**Discussion by EmperorOrokuSaki** [fleekxyz/non-fungible-apps#129] [General]
The only advantage that Counters brings to the contract over basic arithmetic is the checks it does on decrementation. Since the release of Solidity v0.8, the overflow checks have become part of the language itself.
There does not seem to be a reason to keep the library, especially because we don't decrement our counters ever in the contract.

**Comment by zoruka** [comment]
I'm up to not use it if cleans up the code.

**Comment by EmperorOrokuSaki** [comment]
Decided to remove it per our discussion on Discord's daily sync.

**Comment by EmperorOrokuSaki** [comment]
Pull Request: #137

**Reply by EmperorOrokuSaki** [reply]
Will lock this conversation.

**Comment by EmperorOrokuSaki** [comment]
Update: Closing the discussion due to inactivity.
